sdk/src/main/java/io/opentelemetry/sdk/metrics/AbstractBoundInstrument.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import io.opentelemetry.metrics.LabelSet;
+import java.util.concurrent.atomic.AtomicLong;
+
+abstract class AbstractBoundInstrument {
+  private final LabelSet labels;
+  // Atomically counts the number of references (usages) while also keeping a state of
+  // mapped/unmapped into a registry map.
+  private final AtomicLong refCountMapped;
```

Comment:
I would guess that an AtomicInteger would be plenty for this.

Reply:
Are you worried about performance/memory?

Reply:
A tiny bit, yes. atomic integer updates are much easier, since they don't require multiple words on a 32 bit JVM. I don't know how much we care about that, but it's something to consider. And, I sure hope we don't have more than MAX_INT instruments in a single process.  :)

It's a very small thing though; I wouldn't hold up the PR for it.  ;)

Reply:
Please file an issue if this is still a concern :) That change is already merged.

Reply:
I'm not overly concerned. Just thought I'd mention it.  :)